src/parser/parser_driver.cpp

```diff
@@ -12,6 +12,8 @@
 #include "yaramod/types/token_type.h"
 #include "yaramod/utils/filesystem_operations.h"
 
+#include <iostream>
```

Comment:
I started printing through `std::cerr` instead of throwing `ParserError` when parsing fails in `IncompleteMode`.This only happens if a YARA rule is not finished (missing `}`, etc.). It prints the error as Warning but we can still access the `TokenStream` as we can see in `ParseIncompleteRuleNotFinished` cpp parser test.

Reply:
Printing from libraries is generally bad practice. It's much better if library returns the errors/warnings to the user and they can decide what to do with them. Imagine a situation that you don't want to print to `stderr` or simply `stderr` is missing because the application is daemon. How would you know about such error/warning?